gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpRequestFormatter.java

```diff
@@ -30,22 +30,23 @@
 package com.google.api.gax.httpjson;
 
 import com.google.api.core.BetaApi;
-import com.google.gson.Gson;
 import java.util.List;
 import java.util.Map;
-import java.util.Set;
 
-/** Interface for classes that create parts of Http requests from a parameterized message. */
+/** Interface for classes that create parts of HTTP requests from a parameterized message. */
 @BetaApi
-public interface HttpRequestFormatter<MessageFormatT> {
+public interface HttpRequestFormatter<RequestMessageT> {
   /**
    * Return a map where each entry is the name of a query param mapped to the values of the param.
    */
-  Map<String, List<String>> getQueryParams(MessageFormatT apiMessage, Set<String> paramNames);
-
-  /** Return a map where each entry is the name of a path param mapped to the value of the param. */
-  Map<String, String> getPathParams(MessageFormatT apiMessage, String resourceNameField);
+  Map<String, List<String>> getQueryParams(RequestMessageT apiMessage);
 
   /** Write out the inner request body of the given message. */
-  void writeRequestBody(MessageFormatT apiMessage, Gson marshaller, Appendable writer);
+  void writeRequestBody(RequestMessageT apiMessage, Appendable writer);
+
+  /* Return the relative URL path created from the path parameters from the given message. */
+  String getEndpointRelativePath(RequestMessageT apiMessage);
```

Comment:
Maybe we can just call this `getPath` - I think that "path" is generally taken to mean what comes after the domain. ( https://www.ibm.com/support/knowledgecenter/en/SSGMCP_5.1.0/com.ibm.cics.ts.internet.doc/topics/dfhtl_uricomp.html )

Reply:
Sure.